### PR TITLE
Posts: Don't wait for images to load when picking a feature

### PR DESCRIPTION
**Pull request by blowery**
Filter based on the attributes on the dom node instead of trying to load them and then filtering.

### CODE REVIEW

**Comment by blowery** [comment]
This patch removes the requirement to load all of the images in a post before selecting a candidate for the featured image. We originally had this requirement for a couple reasons: some images that could be featured may be missing width and height attributes, and images in the post may not load. Rather than feature a busted image, we'd wait until we had a good one to show.

Unfortunately, this delays adding the featured image to the post card in some instances, which can cause the stream to jump around when loading cards, especially on slower connections.

With this applied, we don't wait on the images to load, we just use the best guess at dimensions we have from the image width and height attributes and possibly the attachment information (only used for featured images currently). This removes the delay and lets us put a placeholder in before the image even tries to load, which prevents the feed from jumping.

To test, pull up the posts stream inside My Sites and scroll around. You shouldn't see any jumping as images load.

File: client/lib/posts/utils.js [hunk 1]
```diff
@@ -151,10 +151,11 @@ var utils = {
 	},
 
 	normalizeAsync: function( post, callback ) {
+
 		postNormalizer(
 			post,
 			[
-				postNormalizer.waitForImagesToLoad,
+				//postNormalizer.waitForImagesToLoad,
```
Comment (gwwar) [inline client/lib/posts/utils.js]: <kbd>delete</kbd>!!

Comment (blowery) [inline client/lib/posts/utils.js]: really we can kill the "async" rules all together and move keepValid over to the sync ruleset.

**Comment by gwwar** [comment]
:+1: This seems like a reasonable tradeoff, and appears to behave pretty well. :ship:

**Comment by blowery** [comment]
Hold on one sec, I can simplify it :)

**Comment by blowery** [comment]
Hrm, actually no I can't. At least not quickly. 

I think we can pull the entire posts-images-store...

File: client/lib/posts/utils.js (hunk 1 above)
Comment (blowery) [inline client/lib/posts/utils.js]: delete'd the line :)

**Comment by blowery** [comment]
squashed in removal and the lint fix. waiting on a retest

**Pull request merged**